### DISCUSSION TITLE
Ryzen 4000 series APU support pt2

### DISCUSSION THREAD

**Discussion by BananaWorks07** [ROCm/ROCm#3878] [General]
recently I noticed that opencl-amd (from the AUR) version 22.10 works remarkably well, but my vram is 
`  Global memory size                              536870912 (512MiB)`
I want more, 512 MB is not a lot, so I was wondering if rocm could support using the gtt memory space as "vram"

**Comment by lanwatch** [comment]
Related to https://github.com/RadeonOpenCompute/ROCm/issues/2014

**Comment by BananaWorks07** [comment]
> Related to #2014

that's not just related, that's an identical issue (probably even a duplicate)
I doubt AMD are ever going to support these processors since they didn't give a shit and closed my issue in 2021 as well

welcome to ROCm, my friend

**Comment by winstonma** [comment]
@Etaash-mathamsetty [UniversalAMDFormBrowser](https://github.com/DavidS95/Smokeless_UMAF) and followed [this video](https://youtu.be/0jwrWCF5fhc) to config the BIOS to get more GPU dedicated memory.

**Comment by BananaWorks07** [comment]
> @Etaash-mathamsetty [UniversalAMDFormBrowser](https://github.com/DavidS95/Smokeless_UMAF) and followed [this video](https://youtu.be/0jwrWCF5fhc) to config the BIOS to get more dedicated memory.

it's too late now, I have a PC with an RX 6800, and ROCm still sucks and doesn't work properly, it is what it is
I will definitely use that to overclock my laptop though

**Comment by winstonma** [comment]
I wrote a [guide](https://github.com/RadeonOpenCompute/ROCm/issues/2227#issuecomment-1639262658) on how to check Pytorch is using GPU. See if that helps.

The most important thing is, as long as you get the `True` result from the following command line then you would be good to go.

```bash
# To check Pytorch is using GPU
$HSA_OVERRIDE_GFX_VERSION=10.3.0 python -c 'import torch; print(torch.cuda.is_available())'
True
```

**Comment by abhimeda** [comment]
@Etaash-mathamsetty Hi, is your issue resolved on the latest ROCm? If so can we close this ticket?

**Comment by BananaWorks07** [comment]
> @Etaash-mathamsetty Hi, is your issue resolved on the latest ROCm? If so can we close this ticket?

If the other issue is anything to go with then no, but I'm not testing this since I don't use an integrated GPU anymore and frankly can't bother when rusticl exists.

**Comment by winstonma** [comment]
@Etaash-mathamsetty based on the kernel message GTT is available in driver level (e.g. some developer can modify PyTorch to use GTT memory). I guess it prove that GTT memory is working. However the problem now is the support of the applications.

You can hack your BIOS to get more video memory (not GTT) so application would work. At the same time you need to wait for the application developer to add GTT support.

**Comment by BananaWorks07** [comment]
> @Etaash-mathamsetty based on the kernel message GTT is available in driver level (e.g. some developer can modify PyTorch to use GTT memory). I guess it prove that GTT memory is working. However the problem now is the support of the applications.
> 
> You can hack your BIOS to get more video memory (not GTT) so application would work. At the same time you need to wait for the application developer to add GTT support.

I think it's poor driver design if the app developer has to include it, i specifically only use rocm for opencl and I'm 90% sure opencl doesn't allow apps to decide which memory heap to put data on. Mesa Vulkan and opengl and opencl drivers report increased vram for example

**Comment by winstonma** [comment]
> > @Etaash-mathamsetty based on the kernel message GTT is available in driver level (e.g. some developer can modify PyTorch to use GTT memory). I guess it prove that GTT memory is working. However the problem now is the support of the applications.
> > 
> > You can hack your BIOS to get more video memory (not GTT) so application would work. At the same time you need to wait for the application developer to add GTT support.
> 
> I think it's poor driver design if the app developer has to include it, i specifically only use rocm for opencl and I'm 90% sure opencl doesn't allow apps to decide which memory heap to put data on. Mesa Vulkan and opengl and opencl drivers report increased vram for example 

I am not sure who is responsible but I asked this question to the PyTorch ROCm team. They said they need to add GTT support. Although I personally think if GTT memory allocation should be done on driver level then application no need to modify (it just work as long as it support ROCm). But currently this is not the case.

**Comment by BananaWorks07** [comment]
> > > @Etaash-mathamsetty based on the kernel message GTT is available in driver level (e.g. some developer can modify PyTorch to use GTT memory). I guess it prove that GTT memory is working. However the problem now is the support of the applications.
> > > 
> > > You can hack your BIOS to get more video memory (not GTT) so application would work. At the same time you need to wait for the application developer to add GTT support.
> > 
> > I think it's poor driver design if the app developer has to include it, i specifically only use rocm for opencl and I'm 90% sure opencl doesn't allow apps to decide which memory heap to put data on. Mesa Vulkan and opengl and opencl drivers report increased vram for example 
> 
> I am not sure who is responsible but I asked this question to the PyTorch ROCm team. They said they need to add GTT support. Although I personally think if GTT allocation is done on driver level then application no need to modify. But currently this is not the case.

Whole point of this issue is to make that the case, because that's how it is on windows unless AMD decided to release drivers that are litterally worse than their original ones with the swap to rocm

**Comment by winstonma** [comment]
Well... I am not sure but 

1. Windows doesn't have the PyTorch that can use ROCm. So ROCm on Windows is sort of broken
2. Windows game can assign the GTT memory
 
Here is my console output from Ubuntu, running ROCm 6.0 driver on my Ryzen 6800U
```
$ sudo dmesg | grep amdgpu
...
[    3.627547] amdgpu 0000:03:00.0: amdgpu: GART: 1024M 0x0000000000000000 - 0x000000003FFFFFFF
[    3.627692] [drm] amdgpu: 512M of VRAM memory ready
[    3.627694] [drm] amdgpu: 7614M of GTT memory ready.
[    3.628651] amdgpu 0000:03:00.0: amdgpu: Will use PSP to load VCN firmware
...
```

So I believe GTT can be used. But the dynamic allocation part doesn't happen on Linux application. I am not sure if that fulfill your definition of _support_ but I just try to explain things. If you want more detailed explanation / explanation why the driver doesn't designed as you expected, please ask @abhimeda

**Comment by Ristovski** [comment]
@Etaash-mathamsetty I suggest you read issue https://github.com/ROCm/ROCm/issues/2014 and the comments.

**Comment by BananaWorks07** [comment]
> @Etaash-mathamsetty I suggest you read issue https://github.com/ROCm/ROCm/issues/2014 and the comments.

I already have

**Comment by winstonma** [comment]
> @Etaash-mathamsetty I suggest you read issue #2014 and the comments.

@Ristovski I am not sure which part talk about APU support. I think it is okay if AMD Driver Team said GTT is supported, and upper level application need to modified to get end-to-end support. But it was never specifically mentioned.

Also I think @Etaash-mathamsetty concern is the understanding that the driver should do the GTT allocation (e.g. I don't need a AMD-Supported Windows DirectX to use GTT memory on my system, it's driver who allocate the memory itself). Hope I understand your concern correctly.
